sleepecg/heartbeat_detection.py

```diff
@@ -63,12 +69,26 @@ def detect_heartbeats(ecg: np.ndarray, fs: float) -> np.ndarray:
         ECG signal.
     fs : float
         Sampling frequency in Hz.
+    backend : {'c', 'numba', 'py'}
+        Which implementation of the squared moving integration and
+        thresholding algorithm to use, by default 'c'.
 
     Returns
     -------
     heartbeat_indices: np.ndarray
         Indices of detected heartbeats.
     """
+    if backend not in _backends:
+        for alternative in ('c', 'numba', 'py'):
+            if alternative in _backends:
+                warnings.warn(
+                    f'Invalid backend for heartbeat_detection: {backend!r}, '
+                    f'using {alternative!r} instead.\n'
+                    f'Possible options are: {_backends}.',
+                )
+                backend = alternative
+                break
```

Comment:
Not sure if we should throw an error if a backend that's not in the list was provided, and only warn and select a fallback backend if either `c` or `numba` was provided but not available.

Reply:
Good idea, much cleaner now :)